VoodooSMBus/VoodooSMBusControllerDriver.cpp

```diff
@@ -365,5 +376,8 @@ IOReturn VoodooSMBusControllerDriver::transferGated(VoodooSMBusControllerMessage
             break;
     }
     
+    if (res < 0)
+        return res;
```

Comment:
Could we just return always `res` here?

Reply:
Actually I think this is never pased through because of how `return command_gate->runAction` works...

Reply:
The return value should be passed through. From IOCommandGate.h
`@result The return value of action if it was called, kIOReturnBadArgument if action is not defined, kIOReturnAborted if a disabled command gate is free()ed before being reenabled.`
Yeah, could just return res I guess. Though transfer doesn't return an IOReturn, just s32. Looking though, kIOReturnSuccess is 0, so functionality should be the same.

Reply:
Interesting, because I looked it up on https://developer.apple.com/documentation/kernel/iocommandgate/1811576-runaction?language=objc and it says something different.

Reply:
👀 I was definitely seeing the behaviour I listed when testing the kext, since we were not reading succesfully in OpenCore at first.

Reply:
i guess there doc online is wrong, then let's just return alsways `res`